### PR TITLE
feat(rust, python): Added from_str_radix method to StringNameSpace that allows to parse strings from any base to i32

### PR DESCRIPTION
**Pull request by cojmeister**
Relates to #6533 

Changes:

- Added from_str_radix(radix: Option<u32>) method to string namespace
- Added relevant test
- Added relevant features to Cargo.toml file

Honestly, I have no idea what I'm doing. In practice it works.  But:

- I'm not sure the tests were placed in the right place, where should I put them?
- Should I set this to be only available on a specific feature? If so, in which file do I to this?
- Should I change back to Cargo.toml file?

Any help and advices are welcome! It's my first attempt at rust proggraming!

thanks!

### CODE REVIEW

File: polars/Cargo.toml [hunk 1]
```diff
@@ -70,6 +72,8 @@ performant = [
   "cse",
   "polars-ops/performant",
   "streaming",
+  "string_encoding",
```
Comment (ritchie46) [inline polars/Cargo.toml:75]: Why do these have to be added to `performant`?

File: polars/Cargo.toml [hunk 2]
```diff
@@ -25,6 +25,8 @@ default = [
   "temporal",
   "fmt",
   "dtype-slim",
+  "string_encoding",
```
Comment (ritchie46) [inline polars/Cargo.toml:28]: These should not be added to `default`

File: polars/tests/it/lazy/expressions/apply.rs [hunk 3]
```diff
@@ -105,3 +105,27 @@ fn test_apply_groups_empty() -> PolarsResult<()> {
 
     Ok(())
 }
+
+#[test]
+fn test_format_str() {
```
Comment (ritchie46) [inline polars/tests/it/lazy/expressions/apply.rs:110]: Can this test be moved to the python test?

**Review by ritchie46 — COMMENTED** [review]
Thanks for the PR. It looks good overal :+1: . I have left a few comments.

File: polars/tests/it/lazy/expressions/apply.rs (hunk 3 above)
Comment (cojmeister) [reply polars/tests/it/lazy/expressions/apply.rs:110]: It could. But in order to do that, shouldn't I port the feature to python as well?

Comment (ritchie46) [reply polars/tests/it/lazy/expressions/apply.rs:110]: Yes, but that is also required for all our functionality. :)

**Comment by cojmeister** [comment]
@ritchie46 I think now it is ready!

File: polars/polars-lazy/polars-plan/src/dsl/function_expr/schema.rs [hunk 4]
```diff
@@ -139,6 +139,7 @@ impl FunctionExpr {
                     Uppercase | Lowercase | Strip(_) | LStrip(_) | RStrip(_) => {
                         with_dtype(DataType::Utf8)
                     }
+                    FromRadix { .. } => with_dtype(DataType::Int64),
```
Comment (ritchie46) [inline polars/polars-lazy/polars-plan/src/dsl/function_expr/schema.rs:142]: The output type will be `Int32`.

**Review by ritchie46 — COMMENTED** [review]
Thanks @cojmeister. The code looks good. I have only one comment because we return the wrong datatype and polars must know it's dtype at all time.

I also think we should not compile this functionality for all polars users. Can you add a feature flag `string_from_radix`?

See features `concat_str` and `str_justify` how this can be done.

**Comment by cojmeister** [comment]
> Thanks @cojmeister. The code looks good. I have only one comment because we return the wrong datatype and polars must know it's dtype at all time.
> 
> I also think we should not compile this functionality for all polars users. Can you add a feature flag `string_from_radix`?
> 
> See features `concat_str` and `str_justify` how this can be done.

@ritchie46 

Done!

**Comment by ritchie46** [comment]
> Done! Great! Going in...

**Pull request merged**